.github/workflows/source_aws.yml

```diff
@@ -88,39 +100,49 @@ jobs:
       - name: Run sync
         if: startsWith(github.head_ref, 'release-please--branches--main--components')
         run: cloudquery sync test/sanity.yml --log-console
+      - name: Save build cache
+        if:   github.ref == 'refs/heads/main'
```

Comment:
What's the reason we only save the cache on the `main` branch?

Reply:
I recommend reading https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache as PRs cannot modify the `main` branch cache anyway.
They can only access the `main` branch cache or their own

Reply:
It will be propagated to the other branches if it's created on `main`.
The other branches caches are only visible to the children branches, so of no use to the most of the PRs.

Butthey will occupy the cache space & we will overflow the 10GiB limit on the caches).

Reply:
But won't that mean PRs almost always use a dirty cache? Let's say someone merges an unrelated PR to `main`, so that means that if I push a new commit to my PR, I'll start using the cache from `main` instead of my PR's cache

Reply:
Regarding cache limits, we're already way over https://github.com/cloudquery/cloudquery/actions/caches, but GitHub evicts the least recently used ones, so PR caches gets evicted fast

Reply:
removed